### ISSUE TITLE
vue/script-indent does not parse enhanced object literal properties correctly

### ISSUE THREAD

**Issue by mrbrendanjs** [vuejs/eslint-plugin-vue#347]
**Tell us about your environment**

* **ESLint Version:** 4.11.0
* **eslint-plugin-vue Version:** 4.2.0
* **Node Version:** 6.11.4

**Please show your full configuration:**

```
module.exports = {
  extends: [
    // 'plugin:vue/recommended',
    'plugin:vue/base',
    '../../../.eslintrc.js',
  ],
  rules: {
    // <script> tag content in vue files should follow our indentation standard
    'vue/script-indent': [
      'error',
      2,
      {
        baseIndent: 1,
        switchCase: 0,
        ignores: [],
      },
    ],
  },
  plugins: [
    'vue',
  ],
};
```

**What did you do? Please include the actual source code causing the issue.**

```vue
<script type="text/babel">
  import fieldBlockList from '../../../components/field-block/list';
  import nameListItem from './name-list-item';
  import vForUniqueKey from '../../../mixins/v-for-unique-key';
  import listMutator from './data/name-list-mutator';

  module.exports = {
    name: 'contact-group-name',
    mixins: [vForUniqueKey],
    components: {
      fieldBlockList,
      nameListItem, // Expected indentation of 8 spaces, but found 6
      vForUniqueKey, // Expected indentation of 10 spaces, but found 6
      listMutator, // Expected indentation of 12 spaces, but found 6
    }, // Expected indentation of 14 spaces, but found 6
    data() {
      return {
        label: 'Name',
        groupName: 'person_name',
        items: listMutator.parseInitial(FW_DATA.person_names),
        listMutator: listMutator,
      };
    },
  };
</script>
```

**What did you expect to happen?**
The indentation, as written above, is correct, and as such i should not be getting any errors

**What actually happened? Please include the actual, raw output from ESLint.**
The errors shown in comments above were posted after linting.
```
line 12, col 1, Error - Expected indentation of 8 spaces but found 6 spaces. (vue/script-indent)
line 13, col 1, Error - Expected indentation of 10 spaces but found 6 spaces. (vue/script-indent)
line 14, col 1, Error - Expected indentation of 12 spaces but found 6 spaces. (vue/script-indent)
line 15, col 1, Error - Expected indentation of 14 spaces but found 4 spaces. (vue/script-indent)
```

To satisfy this rule, I have to continually indent subsequent properties by 1 multiplier, and the final closing mustache one additional multiplier after the last property as such:
```vue
<script type="text/babel">
  import fieldBlockList from '../../../components/field-block/list';
  import nameListItem from './name-list-item';
  import vForUniqueKey from '../../../mixins/v-for-unique-key';
  import listMutator from './data/name-list-mutator';


  module.exports = {
    name: 'contact-group-name',
    components: {
      fieldBlockList,
        nameListItem,
          vForUniqueKey,
            listMutator,
              },
    data() {
      . . .
    };
    },
  };
</script>
```

**Comment by mrbrendanjs** [comment]
I've also found, and I believe it's probably related, that this indent rule does not like to keep hanging indents  consistent when using newlines to chain accessors. eg:
```
axios.get('path/to/resource')
  .then((result) => {
    . . . 
  }); // Error: expected 0 spaces, but found 2
```
The closing brace/paren should line with the call to `.then`, and this is fine with the ESLint indent rule, but the vue/indent-script rule doesn't like it, and wants to remove that indent.

**Comment by mysticatea** [comment]
Thank you for this report.

This seems a duplicate of #344 and we have a patch #346 already. Please track those.
